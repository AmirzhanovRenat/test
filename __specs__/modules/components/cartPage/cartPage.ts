import { CartList } from '@Components/cartPage/cartList/cartList';
import { Container } from '@Core/container';
import { CartForm } from './cartForm/cartForm';

const SELECTORS = {
    addCartItemButton: './/button[contains(text(), "Add Cart Item")]',
    cartList: './/div[@class="cart__list"]',
    form: '[data-testid="form"]',
};

export class CartPageContainer extends Container {
    public async fulfill(initialState = {}): Promise<void> {
        await super.fulfill(initialState);
    }

    public async clickAddCartItemButton(): Promise<void> {
        await document.clickByXpath(SELECTORS.addCartItemButton);
    }

    public async getCartList(): Promise<CartList> {
        const [cartListElement] = await document.waitForXpath(SELECTORS.cartList);
        const cartList = new CartList(cartListElement);
        return cartList;
    }

    public async getForm(): Promise<CartForm> {
        const [formElement] = await document.waitForQuerySelector(SELECTORS.form);
        const form = new CartForm(formElement);
        return form;
    }
}
