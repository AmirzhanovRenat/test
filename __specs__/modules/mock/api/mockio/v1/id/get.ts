import { MockObject } from '@Mocks/mockObject';

export class GetCartItemsMock extends MockObject {
    public constructor() {
        super();
        this.path = {
            url: 'https://mocki.io/v1/a92eda42-69d0-43c5-bc66-cd56e4b8c906',
            method: 'get',
        };
    }

    public getFixture(): Record<string, unknown>[] {
        return [
            {
                id: 1,
                name: 'Some weird name',
                price: 1000,
                quantity: 1,
            },
            {
                id: 2,
                name: 'This should`n be in snapshot!!',
                price: 42,
                quantity: 1,
            },
            {
                id: 3,
                name: 'Wrong Name',
                price: 100,
                quantity: 2,
            },
        ];
    }
}
