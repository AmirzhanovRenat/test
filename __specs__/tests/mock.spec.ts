import { CartPageContainer } from '@Components/cartPage/cartPage';
import { Mock } from '@Core/mock';
import { GetCartItemsMock } from '@Mocks/api/mockio/v1/id/get';

describe('Items Request', () => {
    const mock = Mock.getInstance();
    let cartPage: CartPageContainer;

    beforeEach(async () => {
        cartPage = new CartPageContainer();
        mock.addMocks(new GetCartItemsMock());

        await cartPage.fulfill();
    });

    test('items info should be correct', async () => {
        const cartList = await cartPage.getCartList();
        const items = await cartList.getCartItems();

        for (const item of items) {
            expect(await item.getInfo()).toMatchSnapshot();
        }
    });
});
