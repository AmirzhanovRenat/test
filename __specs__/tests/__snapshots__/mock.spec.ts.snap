// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items Request items info should be correct 1`] = `
Object {
  "name": "SanDisk 128GB Ultra MicroSDXC",
  "price": undefined,
  "priceForAll": NaN,
  "quantity": 1,
}
`;

exports[`Items Request items info should be correct 2`] = `
Object {
  "name": "BENGOO G9000 Stereo Gaming Headset",
  "price": undefined,
  "priceForAll": 42,
  "quantity": 1,
}
`;

exports[`Items Request items info should be correct 3`] = `
Object {
  "name": "Logitech C922x Pro Stream Webcam",
  "price": 100,
  "priceForAll": 200,
  "quantity": 2,
}
`;
